app/ssttpdirectdebit/DirectDebitForm.scala

```diff
@@ -24,31 +24,27 @@ import uk.gov.hmrc.selfservicetimetopay.models.ArrangementDirectDebit.cleanSortC
 import scala.util.Try
 
 object DirectDebitForm {
-  val minAccountNumber = 0
-  val maxAccountNumber = 999999999
-
-  def parseIntOption(str: String) = Try(str.toInt).toOption
 
   def condTrue(condition: Boolean, statement: Boolean) = if (condition) statement else true
 
   val directDebitMapping = mapping(
     "accountName" -> text.verifying("ssttp.direct-debit.form.error.accountName.required", _.trim != "")
-      .verifying("ssttp.direct-debit.form.error.accountName.check", x => condTrue(x.trim != "", x.trim.length > 1))
-      .verifying("ssttp.direct-debit.form.error.accountName.check", x => condTrue(x.length > 1, x.matches("^[a-zA-Z].{1,39}$")))
-      .verifying("ssttp.direct-debit.form.error.accountName.not-text", x => condTrue(x.matches("^[a-zA-Z].{1,39}$"), x.length == x.replaceAll("[^a-zA-Z '.& \\/]", "").length)),
+      .verifying("ssttp.direct-debit.form.error.accountName.check", x => condTrue(x.trim != "", x.trim.length <= 70))
+      .verifying("ssttp.direct-debit.form.error.accountName.check", _.matches("^[a-zA-Z '.&/]{1,39}$")), // regex from API#1856
     "sortCode" -> text
       .verifying("ssttp.direct-debit.form.error.sortCode.required", _.trim != "")
-      .verifying("ssttp.direct-debit.form.error.sortCode.not-valid", isValidSortCode _),
+      .verifying("ssttp.direct-debit.form.error.sortCode.not-valid", x => condTrue(x.trim != "", isValidSortCode(x))),
     "accountNumber" -> text.verifying("ssttp.direct-debit.form.error.accountNumber.required", _.trim != "")
-      .verifying("ssttp.direct-debit.form.error.accountNumber.not-valid", x => (x.trim == "") | (validateNumberLength(x, x.length) && validateNumberLength(x, 8)))
+      .verifying("ssttp.direct-debit.form.error.accountNumber.not-valid", x => (x.trim == "") | (isValidNumber(x, x.length) && validateNumberLength(x)))
   )({ case (name, sc, acctNo) => ArrangementDirectDebit(name, cleanSortCode(sc), acctNo) }
     )({ case arrangementDirectDebit => Some((arrangementDirectDebit.accountName, arrangementDirectDebit.sortCode, arrangementDirectDebit.accountNumber)) })
 
   def isValidSortCode(sortCode: String): Boolean = {
-    hasValidSortCodeCharacters(sortCode) && has6Numbers(sortCode)
+    hasValidSortCodeCharacters(sortCode) && hasSixDigits(sortCode)
```

Comment:
👍

Reply:
I'm a bit puzzled why there isn't just a simple regex for `sortcode`
something like:
```scala
validSortCodeRegex = """^[0-9]{6}$"""
```

Reply:
and AccountNumber
```code
validSortCodeRegex = """^[0-9]{6, 8}$"""
```